src/pages/AddItemForm.js

```diff
@@ -0,0 +1,109 @@
+    const docRef = await addDoc(collection(db, 'groceries'), {
+      item_name: item_name,
+      purchase_interval: purchase_interval,
+      user_token: user_token,
+      last_purchased_date: null,
+    });
+
+    console.log('Document written with ID: ', docRef.id);
+  } catch (e) {
+    console.error('Error adding document: ', e);
+  }
+}
+
+class AddItemForm extends React.Component {
+  constructor(props) {
+    //super is a method from React (create a class from a parent class)
+    super();
+    this.state = {
+      item_name: '',
+      purchase_interval: 7,
+      user_token: props.token.token,
+    };
+    //.bind(this) -The bind() is an inbuilt method in React that is used to pass the data as an argument to the function of a class based component
+    this.handleNameChange = this.handleNameChange.bind(this);
+    this.handleRadioChange = this.handleRadioChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+  //input
+  handleNameChange = (e) => {
+    this.setState({ item_name: e.target.value });
+  };
+  handleRadioChange = (e) => {
+    this.setState({
+      purchase_interval: e.target.value,
+    });
+  };
+  handleSubmit = async (e) => {
+    e.preventDefault();
+    console.log(this.state);
+    addToDb(
+      this.state.item_name,
+      this.state.purchase_interval,
+      this.state.user_token,
+    );
+    alert('Item was submitted: ' + this.state.item_name);
+  };
+  //TODO: check wireframe/accessibility requirements
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label htmlFor="item_name">
+          Item Name:
+          <input
+            type="text"
+            value={this.state.item_name}
+            onChange={this.handleNameChange}
+          />
+        </label>
+        <p>How soon will you buy this again?</p>
+        <fieldset>
+          <label htmlFor="soon">
+            Soon
+            <input
+              checked={this.state.purchase_interval === String(7)}
+              onChange={this.handleRadioChange}
+              type="radio"
+              id="soon"
+              name="purchaseInterval"
+              value="7"
+            />
+          </label>
+          <label htmlFor="kind_of_soon">
+            Kind of Soon
+            <input
+              checked={this.state.purchase_interval === String(14)}
+              onChange={this.handleRadioChange}
+              type="radio"
+              id="kind_of_soon"
+              name="purchaseInterval"
+              value="14"
+            />
+          </label>
+          <label htmlFor="not_soon">
+            Not Soon
+            <input
+              checked={this.state.purchase_interval === String(30)}
+              onChange={this.handleRadioChange}
+              type="radio"
+              id="not_soon"
+              name="purchaseInterval"
+              value="30"
+            />
+          </label>
+        </fieldset>
+        <input type="submit" value="Add Item" />
```

Comment:
It think it may be a little easier to read and have a few advantages to declare this as `<button type="submit">Add Item</button>`

Here's a good little article about it https://thisthat.dev/button-vs-input-type-button/